packages/analytics-browser/src/attribution/campaign-tracker.ts

```diff
@@ -0,0 +1,110 @@
+
+    this.disabled = Boolean(options.disabled);
+    this.trackNewCampaigns = Boolean(options.trackNewCampaigns);
+    this.excludeReferrers = options.excludeReferrers ?? [];
+    if (typeof location !== 'undefined') {
+      this.excludeReferrers.unshift(location.hostname);
+    }
+    this.initialEmptyValue = options.initialEmptyValue ?? EMPTY_VALUE;
+  }
+
+  isNewCampaign(currentCampaign: Campaign, previousCampaign: Campaign) {
+    const isReferrerExcluded = Boolean(
+      currentCampaign.referring_domain && this.excludeReferrers.includes(currentCampaign.referring_domain),
+    );
+    const hasNewUtm =
+      previousCampaign.utm_campaign !== currentCampaign.utm_campaign ||
+      previousCampaign.utm_source !== currentCampaign.utm_source ||
+      previousCampaign.utm_medium !== currentCampaign.utm_medium ||
+      previousCampaign.utm_term !== currentCampaign.utm_term ||
+      previousCampaign.utm_content !== currentCampaign.utm_content;
+    const hasNewReferrer = previousCampaign.referring_domain !== currentCampaign.referring_domain;
+
+    return !isReferrerExcluded && (hasNewUtm || hasNewReferrer);
+  }
+
+  saveCampaignToStorage(campaign: Campaign) {
+    this.storage.set(this.storageKey, campaign);
+  }
+
+  getCampaignFromStorage(): Campaign {
+    return this.storage.get(this.storageKey) || { ...BASE_CAMPAIGN };
+  }
+
+  createCampaignEvent(campaign: Campaign) {
+    const campaignParameters: Campaign = {
+      // This object definition allows undefined keys to be iterated on
+      // in .reduce() to build indentify object
+      ...BASE_CAMPAIGN,
+      ...campaign,
+    };
+    const identifyEvent = Object.entries(campaignParameters).reduce((identify, [key, value]) => {
+      identify.setOnce(`initial_${key}`, value || this.initialEmptyValue);
+      if (value) {
+        return identify.set(key, value);
+      }
+      return identify.unset(key);
+    }, new Identify());
+
+    const pageViewEvent: BaseEvent = {
+      event_type: 'Page View',
+      event_properties: {
+        page_title: /* istanbul ignore next */ (typeof document !== 'undefined' && document.title) || '',
+        page_location: /* istanbul ignore next */ (typeof location !== 'undefined' && location.href) || '',
+        page_path: /* istanbul ignore next */ (typeof location !== 'undefined' && location.pathname) || '',
+      },
+    };
+    return {
+      ...createIdentifyEvent(undefined, undefined, identifyEvent),
+      ...pageViewEvent,
+    };
+  }
+
+  async send(force: boolean) {
+    if (this.disabled) {
+      return;
+    }
+    const currentCampaign = this.parser.parse();
+    const previousCampaign = this.getCampaignFromStorage();
+    if (!force) {
+      if (!this.trackNewCampaigns || !this.isNewCampaign(currentCampaign, previousCampaign)) {
+        return;
+      }
+      this.onNewCampaign(currentCampaign);
+    }
+    await this.track(this.createCampaignEvent(currentCampaign));
+    this.saveCampaignToStorage(currentCampaign);
```

Comment:
updated logic of  whether to track or not. `force` parameter is passed with `isNewSession` from client